sources/common/inc/ISignal.h

```diff
@@ -58,6 +61,10 @@ namespace SIGNAL
                 Signal(name, size, rTimestamp, "wire", sourceHandle),
                 m_Value(value)
             {
+                if (value >= std::pow(2, size))
```

Comment:
Well, I thought the same. But it turns out that for n equaling 64 the expression would have to be broken down. It wouldn't be that simple. Using std::pow() seems more elegant. I brought it back.

Reply:
Got it fixed. It is slightly more complicated than just shifting but these are still basic arithmetic operations and I got rid of std::pow().

Reply:
I pointed that one out not because of cosmetic reasons but because it's wrong.
`std::pow` return type will be `double` in this case and 64bit IEE754 `double` can't accurately represent *every* integer greater than `2^53`. It *could* be fixed with `long double` on some implementations that use 80-bit *extended precision* type which should hold 64 bit integers fine. MSVC **does not** support that.

Well ok, it's expected that you can't have `2^64` in `uint64_t`, since its max value is `2^64 - 1`.
It's not that trivial. What we do want to have is a max number that can be represented by n-bit signal. This is `2^size - 1`. It's just `(1 << size) - 1` except it will fail if `size` is 64 (*undefined behavior*). Because of this, one needs to make sure it will work correctly for 64 (and 0).
Here is [one way to do it](https://stackoverflow.com/questions/52573447/creating-a-mask-with-n-least-significant-bits-set):
```cpp
// Do we check somewhere if size is not greater than 64?
const uint64_t maxSignalValue = (UINT64_C(1) << (size & 63U)) - (size >> 6U) - 1U;
if (value > maxSignalValue) {
// ...
```
That entire "complicated" line could be replaced by `_bzhi_u64(UINT64_C(-1), size);` but since it supported only by certain processors it's not desirable.
See it in action [here](https://wandbox.org/permlink/jybme0OGvYiIxKKD). Notice that `pow` solution **fails** for large numbers.

Reply:
You committed while I was writing above comment. Current version is also incorrect. It will cause undefined behavior if size is equal to 0. But maybe it's checked somewhere else.